### PR TITLE
Warn users when they attempt to create a PR from default branch

### PR DESCRIPTION
**Pull request by vacuus**
Closes https://github.com/sindresorhus/refined-github/issues/1664 .

I think the commit name is pretty self-explanatory. I'm open to any suggestions on the wording of the message. Should we link to a page like this one https://blog.jasonmeridth.com/posts/do-not-issue-pull-requests-from-your-master-branch/ ? It's pretty informative, but I'm worried that this link might break eventually.

 Here's a screenshot of the warning:

screenshot_2019-02-07 sindresorhus refined-github

### CODE REVIEW

**Comment by bfred-it** [comment]
You can replace everything after the first `if` with:

```js
select(/* container selector */).append(
  <div class=“...”> your JSX here </div>
);
```

Also the style should be borrowed from GitHub’s own alerts, by using their classes

**Comment by vacuus** [comment]
Thanks for the feedback! Haven't touched JS in a while so I'm glad you pointed out the issues you found. I had trouble figuring out what class to use for the warning (surprise the `warning` class is already style by githhub).

Here's what it looks like after making the changes you mentioned:

screenshot_2019-02-08 sindresorhus refined-github

I think the warning div looks a little out of place though (could probably use some padding and a border to fit in with the rest of the page). I'm not familiar with github's styling and I'm not sure if there's a better way to style this (without introducing any extra CSS). Any idea if there are already some github style I could leverage?

**Comment by bfred-it** [comment]
Found the right style on `https://github.com/<user>/<repo>/test/settings/installations`

```html
<div class="flash flash-error my-3">
    <strong>Note:</strong>
    <a href="https://developer.github.com/changes/2018-04-25-github-services-deprecation/">GitHub
    Services have been deprecated</a>. Please contact your integrator for more
    information on how to migrate or replace a service with
    <a href="https://developer.github.com/webhooks/">webhooks</a> or
    <a href="https://developer.github.com/apps/differences-between-apps/#about-github-apps/">GitHub Apps</a>.
  </div>
```

File: source/features/warn-pr-from-master.tsx [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+import {React} from 'dom-chef/react';
+import select from 'select-dom';
+import features from '../libs/features';
+import {getUsername} from '../libs/utils';
+import getDefaultBranch from '../libs/get-default-branch';
+
+async function init() {
+	const defaultBranch = await getDefaultBranch();
+	if (!location.pathname.endsWith(`${getUsername()}:${defaultBranch}`)) {
+		return;
```
Comment (bfred-it) [inline source/features/warn-pr-from-master.tsx]: This should return `false`

**Comment by bfred-it** [comment]
I made some changes on the fly, not sure if it's fully correct.

**Comment by bfred-it** [comment]
screenshot 2019-02-11 at 09 55 01

**Comment by bfred-it** [comment]
This is another possible view that needs to be covered, under two different paths:

1. https://github.com/bfred-it/select-dom/compare/ts-html-element-type?expand=1 <br> Users will end up here when clicking on PR suggestions
2. https://github.com/bfred-it/select-dom/compare/master...ts-html-element-type?quick_pull=1 <br> Users will end up here when editing a file while creating a fork... maybe

**Comment by bfred-it** [comment]
Excellent! Thanks @vacuus 🍏

**Pull request merged**